docs/wiki/EL/evm.md

```diff
@@ -0,0 +1,623 @@
+
+![EVM stack](../../images/evm/stack.gif)
+
+> The EVM stack has a maximum size of 1024 items.
+
+During bytecode execution, EVM stack functions as a _scratchpad_: opcodes consume data from the top and push results back (See the `ADD` opcode below). Consider a simple addition program:
+
+![EVM stack addition](../../images/evm/stack-addition.gif)
+
+Reminder: All values are in hexadecimal, so `0x06 + 0x07 = 0x0d (decimal: 13)`.
+
+Let's take a moment to celebrate our first EVM assembly code 🎉.
+
+## Program counter
+
+In the example above, the values on the left of the assembly code represent the byte offset (starting at 0) of each opcode within the bytecode:
+
+| Bytecode | Assembly | Length of Instruction in bytes | Offset in hex |
+| -------- | -------- | ------------------------------ | ------------- |
+| 60 06    | PUSH1 06 | 2                              | 00            |
+| 60 07    | PUSH1 07 | 2                              | 02            |
+| 01       | ADD      | 1                              | 04            |
+
+EVM **program counter** stores the byte offset of the next opcode (highlighted) to be executed.
+
+![EVM program counter](../../images/evm/program-counter.gif)
+
+The `JUMP` opcode directly sets the program counter, enabling dynamic control flow and contributing to the EVM's [Turing completeness](https://en.wikipedia.org/wiki/Turing_completeness) by allowing flexible program execution paths.
+
+![EVM JUMP opcode](../../images/evm/jump-opcode.gif)
+
+The code runs in an infinite loop, repeatedly adding 7. It introduces two new opcodes:
+
+- **JUMP**: Sets the program counter to stack top value (02 in our case), determining the next instruction to execute.
+- **JUMPDEST**: Marks the destination of a jump operation, ensuring intended destinations and preventing unwanted control flow disruptions.
+
+> High level languages like [Solidity](https://soliditylang.org/) leverage jumps for constructs like if conditions, for loops, and internal functions calls.
+
+## Gas
+
+Our innocent little program may seem harmless. However, infinite loops in EVM pose a significant threat: they can **devour resources**, potentially causing network [**DoS attacks**.](https://en.wikipedia.org/wiki/Denial-of-service_attack)
+
+The EVM's **gas** mechanism tackles such threats by acting as a currency for computational resources. Transactions pay gas in **Ether (ETH)** to use the EVM, and if they run out of gas before finishing (like an infinite loop), the EVM halts them to prevent resource hogging.
+
+This protects the network from getting clogged by resource-intensive or malicious activities. Since gas restricts computations to a finite number of steps, the EVM is considered **quasi Turing complete**.
+
+![EVM Gas](../../images/evm/evm-gas.gif)
+
+Our example assumed 1 unit of gas per opcode for simplicity, but the actual cost varies based on complexity. The core concept, however, remains unchanged.
```

Comment:
you can elaborate a tiny bit here, that the gas cost are measured to reflect the constrains of hardware - storage or computation